extras/src/main/scala/io/odin/extras/derivation/context.scala

```diff
@@ -0,0 +1,78 @@
+package io.odin.extras.derivation
+
+import io.odin.meta._
+import magnolia.{CaseClass, Magnolia, Param, SealedTrait}
+
+object context {
+
+  type Typeclass[A] = ToContextData[A]
+
+  def combine[A](ctx: CaseClass[Typeclass, A]): Typeclass[A] = value => {
+    if (ctx.isValueClass) {
+      ctx.parameters.headOption.fold[ContextData](ContextMap(Map.empty))(param =>
```

Comment:
I think `head` is safe to use in this context, no need for the fold if we know it's a value class